docs/how-to-guides/upgrade-rook-ceph.md

```diff
@@ -0,0 +1,180 @@
+  - [Step 5: Perform updates](#step-5-perform-updates)
+  - [Step 6: Verify that the CSI images are updated](#step-6-verify-that-the-csi-images-are-updated)
+  - [Step 7: Final checks](#step-7-final-checks)
+- [Additional resources](#additional-resources)
+
+## Introduction
+
+Rook Ceph is one of the storage providers of Lokomotive. With a distributed system as complex as
+Ceph, the upgrade process is not trivial. This document enlists steps on how to perform the upgrade
+and how to monitor this process.
+
+## Steps
+
+Following steps are inspired by [`rook`](https://rook.io/docs/rook/master/ceph-upgrade.html) docs.
+
+### Step 1: Ensure `AUTOSCALE` is `on`
+
+Exec into the toolbox pod as specified in [this
+doc](rook-ceph-storage.md#enable-and-access-toolbox). Once you get the shell access, run the
+following command:
+
+```console
+# ceph osd pool autoscale-status | grep replicapool
+POOL           SIZE  TARGET SIZE  RATE  RAW CAPACITY   RATIO  TARGET RATIO  EFFECTIVE RATIO  BIAS  PG_NUM  NEW PG_NUM  AUTOSCALE
+replicapool      0                 3.0         3241G  0.0000                                  1.0      32              on
+```
+
+Ensure that the `AUTOSCALE` column outputs `on` and not `warn`. If the output of the `AUTOSCALE`
+column says `warn`, then run the following command, to make sure that pool autoscaling is enabled.
+This is required to ensure that placement groups scale up as the data in the cluster increases.
+
+```bash
+ceph osd pool set replicapool pg_autoscale_mode on
+```
+
+### Step 2: Watch
+
+Watch events, updates and pods.
+
+#### Step 2.1: Ceph status
+
+Leave the following running in the toolbox pod:
+
+```bash
+watch ceph status
+```
+
+Ensure that the output says that `health:` is `HEALTH_OK`. Match the output such that everything
+looks fine as explained in the [rook upgrade
+docs](https://rook.io/docs/rook/master/ceph-upgrade.html#status-output).
+
+> **IMPORTANT**: Don't proceed further if the output is anything other than `HEALTH_OK`.
+
+#### Step 2.2: Pods in rook namespace
+
+Keep an eye on the pods status in another terminal window from the `rook` namespace. Leave the
+following command running:
+
+```bash
+watch kubectl -n rook get pods -o wide
+```
+
+#### Step 2.3: Rook version update
+
+Run the following command in a new terminal window to keep an eye on the rook version update as it
+is upgrades for all the sub-components:
+
+```bash
+watch --exec kubectl -n rook get deployments -l rook_cluster=rook -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+```bash
+watch --exec kubectl -n rook get jobs -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \tsucceeded: "}{.status.succeeded}{"      \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+You should see that `rook-version` slowly changes to `v1.4.6`.
```

Comment:
Should we mention specific version here?

Reply:
Yep, this is an upgrade doc. So anyone using latest release of lokoctl should expect above version.

Reply:
I thought we make document version-independent. This means whenever we update Rook Ceph, we should update this document as well? This perhaps also apply to #1163 then :thinking:

Reply:
Yep this applies to both the docs.